lib/puppet/util/windows/eventlog.rb

```diff
@@ -31,7 +31,7 @@ class Puppet::Util::Windows::EventLog
   def initialize(source_name = 'Puppet')
     @eventlog_handle = RegisterEventSourceW(FFI::Pointer::NULL, wide_string(source_name))
     if @eventlog_handle == NULL_HANDLE
-      raise EventlogError.new("failed to open Windows eventlog")
+      raise EventLogError.new("RegisterEventSourceW failed to open Windows eventlog (exit code #{FFI.errno})", FFI.errno)
```

Comment:
This seemed like a great solution - but then I tried something out, and I don't think it will quite work out.  Here's why:

`SystemCallError` attempts to look up the given error code and translate it to something in the Ruby library, which won't make sense given Win32 error codes don't map to Ruby errors.  Assume the Win32 error was `5`, which is `Access Denied` in Windows.

This is what Ruby does if you pass the value `5` through to the `SystemCallError` initializer:

```
[1] pry(main)> raise SystemCallError.new('hi', 5)
Errno::EIO: Input/output error - hi
from (pry):1:in `__pry__'
```